ddd-03-fd-blocking.c

```diff
@@ -0,0 +1,188 @@
+#include <openssl/ssl.h>
+
+/* 
+ * Demo 1: Client — Client Creates FD — Blocking
+ * =============================================
+ *
+ * This is an example of (part of) an application which uses libssl in a simple,
+ * synchronous, blocking fashion. The client is responsible for creating the
+ * socket and passing it to libssl. The functions show all interactions with
+ * libssl the application makes, and would hypothetically be linked into a
+ * larger application.
+ */
+
+/*
+ * The application is initializing and wants an SSL_CTX which it will use for
+ * some number of outgoing connections, which it creates in subsequent calls to
+ * new_conn. The application may also call this function multiple times to
+ * create multiple SSL_CTX.
+ */
+SSL_CTX *create_ssl_ctx(void)
+{
+    SSL_CTX *ctx;
+
+    ctx = SSL_CTX_new(TLS_client_method());
+    if (ctx == NULL)
+        return NULL;
+
+    /* Enable trust chain verification. */
+    SSL_CTX_set_verify(ctx, SSL_VERIFY_PEER, NULL);
+
+    /* Load default root CA store. */
+    if (SSL_CTX_set_default_verify_paths(ctx) == 0) {
+        SSL_CTX_free(ctx);
+        return NULL;
+    }
+
+    return ctx;
+}
+
+/*
+ * The application wants to create a new outgoing connection using a given
+ * SSL_CTX.
+ *
+ * hostname is a string like "example.com" used for certificate validation.
+ */
+SSL *new_conn(SSL_CTX *ctx, int fd, const char *bare_hostname)
+{
+    SSL *ssl;
+
+    ssl = SSL_new(ctx);
+    if (ssl == NULL)
+        return NULL;
+
+    SSL_set_connect_state(ssl); /* cannot fail */
```

Comment:
This isn't necessary, since you've used `TLS_client_method` not `TLS_method`. It's harmless though.

Reply:
This isn't matching my experience. If I remove this line, the demo stops working...

Reply:
It still doesn't work for me without `SSL_set_connect_state` and this is on the latest master. How are you building/running? I've added a Makefile, try `make test`.